### ISSUE TITLE
Crash trying to revert after p4 sync in command line

### ISSUE THREAD

**Issue by singalen** [groboclown/p4ic4idea#195]
**Describe the bug**
Plugin crashed when trying to revert locally deleted files from Local Changes view.

** Steps**

1. p4 sync from a command line. Receive some deletions in the latest changes.
2. do NOT refresh the repository.
3. Go to Local changes, see some "files deleted locally". Actually, they were deleted in p4 repo, but plugin doesn't know it yet.
4. Select the "deleted files" tree node in the view.
5. Try to revert the deletions.

**Desktop (please complete the following information):**
 - OS: Mac OS X 10.14
 - IDE and version: PyCharm Community 2018.3.3
 - Version: 0.10.3

Crash stack:

```
java.lang.NullPointerException: element cannot be mapped to a null key
	at java.util.Objects.requireNonNull(Objects.java:228)
	at java.util.stream.Collectors.lambda$groupingBy$45(Collectors.java:907)
	at java.util.stream.ReduceOps$3ReducingSink.accept(ReduceOps.java:169)
	at java.util.ArrayList$ArrayListSpliterator.forEachRemaining(ArrayList.java:1376)
	at java.util.stream.AbstractPipeline.copyInto(AbstractPipeline.java:481)
	at java.util.stream.AbstractPipeline.wrapAndCopyInto(AbstractPipeline.java:471)
	at java.util.stream.ReduceOps$ReduceOp.evaluateSequential(ReduceOps.java:708)
	at java.util.stream.AbstractPipeline.evaluate(AbstractPipeline.java:234)
	at java.util.stream.ReferencePipeline.collect(ReferencePipeline.java:499)
	at net.groboclown.p4plugin.extension.P4RollbackEnvironment.forceSync(P4RollbackEnvironment.java:187)
	at net.groboclown.p4plugin.extension.P4RollbackEnvironment.rollbackMissingFileDeletion(P4RollbackEnvironment.java:132)
	at com.intellij.openapi.vcs.changes.actions.RollbackDeletionAction.processFiles(RollbackDeletionAction.java:32)
	at com.intellij.openapi.vcs.changes.actions.AbstractMissingFilesAction.lambda$null$0(AbstractMissingFilesAction.java:44)
	at java.util.HashMap.forEach(HashMap.java:1288)
	at com.intellij.openapi.vcs.changes.ChangesUtil.processItemsByVcs(ChangesUtil.java:307)
	at com.intellij.openapi.vcs.changes.ChangesUtil.processFilePathsByVcs(ChangesUtil.java:325)
	at com.intellij.openapi.vcs.changes.actions.AbstractMissingFilesAction.lambda$actionPerformed$1(AbstractMissingFilesAction.java:43)
	at com.intellij.openapi.progress.impl.CoreProgressManager$2.run(CoreProgressManager.java:248)
	at com.intellij.openapi.progress.impl.CoreProgressManager$TaskRunnable.run(CoreProgressManager.java:727)
	at com.intellij.openapi.progress.impl.CoreProgressManager$5.run(CoreProgressManager.java:442)
	at com.intellij.openapi.progress.impl.CoreProgressManager.lambda$runProcess$2(CoreProgressManager.java:164)
	at com.intellij.openapi.progress.impl.CoreProgressManager.registerIndicatorAndRun(CoreProgressManager.java:582)
	at com.intellij.openapi.progress.impl.CoreProgressManager.executeProcessUnderProgress(CoreProgressManager.java:532)
	at com.intellij.openapi.progress.impl.ProgressManagerImpl.executeProcessUnderProgress(ProgressManagerImpl.java:87)
	at com.intellij.openapi.progress.impl.CoreProgressManager.runProcess(CoreProgressManager.java:151)
	at com.intellij.openapi.application.impl.ApplicationImpl.lambda$null$10(ApplicationImpl.java:592)
	at com.intellij.openapi.application.impl.ApplicationImpl$1.run(ApplicationImpl.java:314)
	at java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:511)
	at java.util.concurrent.FutureTask.run(FutureTask.java:266)
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1142)
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:617)
	at java.lang.Thread.run(Thread.java:745)
java.lang.NullPointerException: element cannot be mapped to a null key
	at java.util.Objects.requireNonNull(Objects.java:228)
	at java.util.stream.Collectors.lambda$groupingBy$45(Collectors.java:907)
	at java.util.stream.ReduceOps$3ReducingSink.accept(ReduceOps.java:169)
	at java.util.ArrayList$ArrayListSpliterator.forEachRemaining(ArrayList.java:1376)
	at java.util.stream.AbstractPipeline.copyInto(AbstractPipeline.java:481)
	at java.util.stream.AbstractPipeline.wrapAndCopyInto(AbstractPipeline.java:471)
	at java.util.stream.ReduceOps$ReduceOp.evaluateSequential(ReduceOps.java:708)
	at java.util.stream.AbstractPipeline.evaluate(AbstractPipeline.java:234)
	at java.util.stream.ReferencePipeline.collect(ReferencePipeline.java:499)
	at net.groboclown.p4plugin.extension.P4RollbackEnvironment.forceSync(P4RollbackEnvironment.java:187)
	at net.groboclown.p4plugin.extension.P4RollbackEnvironment.rollbackMissingFileDeletion(P4RollbackEnvironment.java:132)
	at com.intellij.openapi.vcs.changes.actions.RollbackDeletionAction.processFiles(RollbackDeletionAction.java:32)
	at com.intellij.openapi.vcs.changes.actions.AbstractMissingFilesAction.lambda$null$0(AbstractMissingFilesAction.java:44)
	at java.util.HashMap.forEach(HashMap.java:1288)
	at com.intellij.openapi.vcs.changes.ChangesUtil.processItemsByVcs(ChangesUtil.java:307)
	at com.intellij.openapi.vcs.changes.ChangesUtil.processFilePathsByVcs(ChangesUtil.java:325)
	at com.intellij.openapi.vcs.changes.actions.AbstractMissingFilesAction.lambda$actionPerformed$1(AbstractMissingFilesAction.java:43)
	at com.intellij.openapi.progress.impl.CoreProgressManager$2.run(CoreProgressManager.java:248)
	at com.intellij.openapi.progress.impl.CoreProgressManager$TaskRunnable.run(CoreProgressManager.java:727)
	at com.intellij.openapi.progress.impl.CoreProgressManager$5.run(CoreProgressManager.java:442)
	at com.intellij.openapi.progress.impl.CoreProgressManager.lambda$runProcess$2(CoreProgressManager.java:164)
	at com.intellij.openapi.progress.impl.CoreProgressManager.registerIndicatorAndRun(CoreProgressManager.java:582)
	at com.intellij.openapi.progress.impl.CoreProgressManager.executeProcessUnderProgress(CoreProgressManager.java:532)
	at com.intellij.openapi.progress.impl.ProgressManagerImpl.executeProcessUnderProgress(ProgressManagerImpl.java:87)
	at com.intellij.openapi.progress.impl.CoreProgressManager.runProcess(CoreProgressManager.java:151)
	at com.intellij.openapi.application.impl.ApplicationImpl.lambda$null$10(ApplicationImpl.java:592)
	at com.intellij.openapi.application.impl.ApplicationImpl$1.run(ApplicationImpl.java:314)
	at java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:511)
	at java.util.concurrent.FutureTask.run(FutureTask.java:266)
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1142)
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:617)
	at java.lang.Thread.run(Thread.java:745)java.lang.NullPointerException: element cannot be mapped to a null key
	at java.util.Objects.requireNonNull(Objects.java:228)
	at java.util.stream.Collectors.lambda$groupingBy$45(Collectors.java:907)
	at java.util.stream.ReduceOps$3ReducingSink.accept(ReduceOps.java:169)
	at java.util.ArrayList$ArrayListSpliterator.forEachRemaining(ArrayList.java:1376)
	at java.util.stream.AbstractPipeline.copyInto(AbstractPipeline.java:481)
	at java.util.stream.AbstractPipeline.wrapAndCopyInto(AbstractPipeline.java:471)
	at java.util.stream.ReduceOps$ReduceOp.evaluateSequential(ReduceOps.java:708)
	at java.util.stream.AbstractPipeline.evaluate(AbstractPipeline.java:234)
	at java.util.stream.ReferencePipeline.collect(ReferencePipeline.java:499)
	at net.groboclown.p4plugin.extension.P4RollbackEnvironment.forceSync(P4RollbackEnvironment.java:187)
	at net.groboclown.p4plugin.extension.P4RollbackEnvironment.rollbackMissingFileDeletion(P4RollbackEnvironment.java:132)
	at com.intellij.openapi.vcs.changes.actions.RollbackDeletionAction.processFiles(RollbackDeletionAction.java:32)
	at com.intellij.openapi.vcs.changes.actions.AbstractMissingFilesAction.lambda$null$0(AbstractMissingFilesAction.java:44)
	at java.util.HashMap.forEach(HashMap.java:1288)
	at com.intellij.openapi.vcs.changes.ChangesUtil.processItemsByVcs(ChangesUtil.java:307)
	at com.intellij.openapi.vcs.changes.ChangesUtil.processFilePathsByVcs(ChangesUtil.java:325)
	at com.intellij.openapi.vcs.changes.actions.AbstractMissingFilesAction.lambda$actionPerformed$1(AbstractMissingFilesAction.java:43)
	at com.intellij.openapi.progress.impl.CoreProgressManager$2.run(CoreProgressManager.java:248)
	at com.intellij.openapi.progress.impl.CoreProgressManager$TaskRunnable.run(CoreProgressManager.java:727)
	at com.intellij.openapi.progress.impl.CoreProgressManager$5.run(CoreProgressManager.java:442)
	at com.intellij.openapi.progress.impl.CoreProgressManager.lambda$runProcess$2(CoreProgressManager.java:164)
	at com.intellij.openapi.progress.impl.CoreProgressManager.registerIndicatorAndRun(CoreProgressManager.java:582)
	at com.intellij.openapi.progress.impl.CoreProgressManager.executeProcessUnderProgress(CoreProgressManager.java:532)
	at com.intellij.openapi.progress.impl.ProgressManagerImpl.executeProcessUnderProgress(ProgressManagerImpl.java:87)
	at com.intellij.openapi.progress.impl.CoreProgressManager.runProcess(CoreProgressManager.java:151)
	at com.intellij.openapi.application.impl.ApplicationImpl.lambda$null$10(ApplicationImpl.java:592)
	at com.intellij.openapi.application.impl.ApplicationImpl$1.run(ApplicationImpl.java:314)
	at java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:511)
	at java.util.concurrent.FutureTask.run(FutureTask.java:266)
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1142)
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:617)
	at java.lang.Thread.run(Thread.java:745)java.lang.NullPointerException: element cannot be mapped to a null key
	at java.util.Objects.requireNonNull(Objects.java:228)
	at java.util.stream.Collectors.lambda$groupingBy$45(Collectors.java:907)
	at java.util.stream.ReduceOps$3ReducingSink.accept(ReduceOps.java:169)
	at java.util.ArrayList$ArrayListSpliterator.forEachRemaining(ArrayList.java:1376)
	at java.util.stream.AbstractPipeline.copyInto(AbstractPipeline.java:481)
	at java.util.stream.AbstractPipeline.wrapAndCopyInto(AbstractPipeline.java:471)
	at java.util.stream.ReduceOps$ReduceOp.evaluateSequential(ReduceOps.java:708)
	at java.util.stream.AbstractPipeline.evaluate(AbstractPipeline.java:234)
	at java.util.stream.ReferencePipeline.collect(ReferencePipeline.java:499)
	at net.groboclown.p4plugin.extension.P4RollbackEnvironment.forceSync(P4RollbackEnvironment.java:187)
	at net.groboclown.p4plugin.extension.P4RollbackEnvironment.rollbackMissingFileDeletion(P4RollbackEnvironment.java:132)
	at com.intellij.openapi.vcs.changes.actions.RollbackDeletionAction.processFiles(RollbackDeletionAction.java:32)
	at com.intellij.openapi.vcs.changes.actions.AbstractMissingFilesAction.lambda$null$0(AbstractMissingFilesAction.java:44)
	at java.util.HashMap.forEach(HashMap.java:1288)
	at com.intellij.openapi.vcs.changes.ChangesUtil.processItemsByVcs(ChangesUtil.java:307)
	at com.intellij.openapi.vcs.changes.ChangesUtil.processFilePathsByVcs(ChangesUtil.java:325)
	at com.intellij.openapi.vcs.changes.actions.AbstractMissingFilesAction.lambda$actionPerformed$1(AbstractMissingFilesAction.java:43)
	at com.intellij.openapi.progress.impl.CoreProgressManager$2.run(CoreProgressManager.java:248)
	at com.intellij.openapi.progress.impl.CoreProgressManager$TaskRunnable.run(CoreProgressManager.java:727)
	at com.intellij.openapi.progress.impl.CoreProgressManager$5.run(CoreProgressManager.java:442)
	at com.intellij.openapi.progress.impl.CoreProgressManager.lambda$runProcess$2(CoreProgressManager.java:164)
	at com.intellij.openapi.progress.impl.CoreProgressManager.registerIndicatorAndRun(CoreProgressManager.java:582)
	at com.intellij.openapi.progress.impl.CoreProgressManager.executeProcessUnderProgress(CoreProgressManager.java:532)
	at com.intellij.openapi.progress.impl.ProgressManagerImpl.executeProcessUnderProgress(ProgressManagerImpl.java:87)
	at com.intellij.openapi.progress.impl.CoreProgressManager.runProcess(CoreProgressManager.java:151)
	at com.intellij.openapi.application.impl.ApplicationImpl.lambda$null$10(ApplicationImpl.java:592)
	at com.intellij.openapi.application.impl.ApplicationImpl$1.run(ApplicationImpl.java:314)
	at java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:511)
	at java.util.concurrent.FutureTask.run(FutureTask.java:266)
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1142)
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:617)
	at java.lang.Thread.run(Thread.java:745)
```

**Comment by groboclown** [comment]
Thanks for the stack trace!  This will make debugging the issue much easier.

**Comment by groboclown** [comment]
This is fixed in v0.10.4.  Thanks again for reporting the issue.
